provider/pkg/provider/provider.go

```diff
@@ -31,12 +30,27 @@ import (
 	pbempty "github.com/golang/protobuf/ptypes/empty"
 )
 
+var PulumiResources = []xyzResource{
+	&xyzRandomResource{},
+}
+
 type xyzProvider struct {
 	host    *provider.HostClient
 	name    string
 	version string
 }
 
+type xyzResource interface {
+	Configure(config xyzConfig)
+	Diff(req *pulumirpc.DiffRequest) (*pulumirpc.DiffResponse, error)
+	Create(req *pulumirpc.CreateRequest) (*pulumirpc.CreateResponse, error)
+	Delete(req *pulumirpc.DeleteRequest) (*pbempty.Empty, error)
+	Check(req *pulumirpc.CheckRequest) (*pulumirpc.CheckResponse, error)
+	Update(req *pulumirpc.UpdateRequest) (*pulumirpc.UpdateResponse, error)
+	Read(req *pulumirpc.ReadRequest) (*pulumirpc.ReadResponse, error)
+	Name() string
```

Comment:
This doesn't look like the full list of methods that need to be implemented. `Invoke`, `StreamInvoke`, and `Attach` are missing for sure.